test/extensions/filters/network/zookeeper_proxy/config_test.cc

```diff
@@ -120,162 +173,101 @@ max_packet_bytes: 1048576
+  EXPECT_EQ(threshold_override.threshold(),
+            ProtobufWkt::util::TimeUtil::MillisecondsToDuration(151));
+
+  Network::FilterFactoryCb cb = factory_.createFilterFactoryFromProto(proto_config_, context_);
+  EXPECT_CALL(connection_, addFilter(_));
+  cb(connection_);
 }
 
-TEST(ZookeeperFilterConfigTest, FullConfig) {
-  const std::string yaml = R"EOF(
-stat_prefix: test_prefix
-max_packet_bytes: 1048576
-default_latency_threshold: "0.1s"
-latency_threshold_overrides:
-  - opcode: Connect
-    threshold: "0.151s"
-  - opcode: Create
-    threshold: "0.152s"
-  - opcode: Delete
-    threshold: "0.153s"
-  - opcode: Exists
-    threshold:
-      nanos: 154000000
-  - opcode: GetData
-    threshold:
-      nanos: 155000000
-  - opcode: SetData
-    threshold:
-      nanos: 156000000
-  - opcode: GetAcl
-    threshold:
-      nanos: 157000000
-  - opcode: SetAcl
-    threshold:
-      nanos: 158000000
-  - opcode: GetChildren
-    threshold:
-      nanos: 159000000
-  - opcode: Sync
-    threshold:
-      nanos: 160000000
-  - opcode: Ping
-    threshold:
-      nanos: 161000000
-  - opcode: GetChildren2
-    threshold:
-      nanos: 162000000
-  - opcode: Check
-    threshold:
-      nanos: 163000000
-  - opcode: Multi
-    threshold:
-      nanos: 164000000
-  - opcode: Create2
-    threshold:
-      nanos: 165000000
-  - opcode: Reconfig
-    threshold:
-      nanos: 166000000
-  - opcode: CheckWatches
-    threshold:
-      nanos: 167000000
-  - opcode: RemoveWatches
-    threshold:
-      nanos: 168000000
-  - opcode: CreateContainer
-    threshold:
-      nanos: 169000000
-  - opcode: CreateTtl
-    threshold:
-      nanos: 170000000
-  - opcode: Close
-    threshold:
-      nanos: 171000000
-  - opcode: SetAuth
-    threshold:
-      nanos: 172000000
-  - opcode: SetWatches
-    threshold:
-      nanos: 173000000
-  - opcode: GetEphemerals
-    threshold:
-      nanos: 174000000
-  - opcode: GetAllChildrenNumber
-    threshold:
-      nanos: 175000000
-  - opcode: SetWatches2
-    threshold:
-      nanos: 176000000
-  )EOF";
-
-  ZooKeeperProxyProtoConfig proto_config;
-  TestUtility::loadFromYamlAndValidate(yaml, proto_config);
-
-  testing::NiceMock<Server::Configuration::MockFactoryContext> context;
-  ZooKeeperConfigFactory factory;
-
-  Network::FilterFactoryCb cb = factory.createFilterFactoryFromProto(proto_config, context);
-  Network::MockConnection connection;
-  EXPECT_CALL(connection, addFilter(_));
-  cb(connection);
+TEST_F(ZookeeperFilterConfigTest, FullConfig) {
+  const ProtobufWkt::EnumDescriptor* opcode_descriptor = envoy::extensions::filters::network::
+      zookeeper_proxy::v3::LatencyThresholdOverride_Opcode_descriptor();
+  std::string yaml = populateFullConfig(opcode_descriptor);
+  TestUtility::loadFromYamlAndValidate(yaml, proto_config_);
+
+  EXPECT_EQ(proto_config_.stat_prefix(), "test_prefix");
+  EXPECT_EQ(proto_config_.max_packet_bytes().value(), 1048576);
```

Comment:
it's hard to tell; are we now covering the setting of all the fields?

Reply:
Yes, we cover all configs in this test case, including
- stat_prefix
- max_packet_bytes
- enable_latency_threshold_metrics
- default_latency_threshold
- 26 latency_threshold_overrides